b2/src/main/java/org/jclouds/b2/handlers/ParseB2ErrorFromJsonContent.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.labs.b2.handlers;
+
+import org.jclouds.http.HttpCommand;
+import org.jclouds.http.HttpErrorHandler;
+import org.jclouds.http.HttpResponse;
+import org.jclouds.http.functions.ParseJson;
+import org.jclouds.json.Json;
+import org.jclouds.labs.b2.B2ResponseException;
+import org.jclouds.labs.b2.domain.B2Error;
+
+import com.google.inject.Inject;
+import com.google.inject.TypeLiteral;
+
+public class ParseB2ErrorFromJsonContent extends ParseJson<B2Error> implements HttpErrorHandler {
+
+   @Inject
+   public ParseB2ErrorFromJsonContent(Json json) {
+      super(json, TypeLiteral.get(B2Error.class));
+   }
+
+   private static Exception refineException(B2Error error, Exception exception) {
+      if ("bad_json".equals(error.code())) {
+         return new IllegalArgumentException(error.message(), exception);
+      } else {
```

Comment:
Let's address this in the upcoming object pull request which will encounter 404 errors.